### DISCUSSION TITLE
export 语法支持“按需”导出集合

### DISCUSSION THREAD

**Discussion by berwin** [JSCIG/es-discuss#42] [Ideas]
有时希望按需选择一些需要导出的元素，并以集合的方式导出。

期待的语法：

```javascript
// a.js 导出
export { name, age } as person from './persion';

// b.js 导入
import { person } from 'a.js';
console.log(person.name, person.age);
```

现在由于没有这样的语法，所以只能无奈将需要按需导出的元素单独拆分到一个文件里。

**Comment by ljharb** [comment]
That doesn't seem like that much of an improvement over:
```js
import { name, age } from './person';
export const person = { name, age }
```

especially since creating the object would have to sever the live bindings, and i expect an export-from syntax to preserve live bindings.

**Comment by hax** [comment]
It seems module declaration proposal might give us a workaround:

```js
module m { export { name, age } from './person' }

import * as person from m
console.log(person.name, person.age);
```
